src/tests/data/restore.test.ts

```diff
@@ -0,0 +1,527 @@
+import * as restore from "../../data/restore";
+import {
+  ExcalidrawTextElement,
+  ExcalidrawFreeDrawElement,
+  ExcalidrawLinearElement,
+  ExcalidrawElement,
+} from "../../element/types";
+import * as sizeHelpers from "../../element/sizeHelpers";
+import { API } from "../helpers/api";
+import { getDefaultAppState } from "../../appState";
+import { ImportedDataState } from "../../data/types";
+import { NormalizedZoomValue } from "../../types";
+
+const mockSizeHelper = jest.spyOn(sizeHelpers, "isInvisiblySmallElement");
+
+beforeEach(() => {
+  mockSizeHelper.mockReset();
+});
+
+describe("restoreElements", () => {
+  it("should return empty array when element is null", () => {
+    expect(restore.restoreElements(null)).toStrictEqual([]);
+  });
+
+  it("should not call isInvisiblySmallElement when element is a selection element", () => {
+    const selectionEl = { type: "selection" } as ExcalidrawElement;
+    const restoreElements = restore.restoreElements([selectionEl]);
+    expect(restoreElements.length).toBe(0);
+    expect(sizeHelpers.isInvisiblySmallElement).toBeCalledTimes(0);
+  });
+
+  it("should return empty array when input type is not supported", () => {
+    const dummyNotSupportedElement = API.createElement({ type: "text" });
+
+    Object.defineProperty(dummyNotSupportedElement, "type", {
+      get: jest.fn(() => "not supported element type"),
+    });
+
+    expect(restore.restoreElements([dummyNotSupportedElement]).length).toBe(0);
+  });
+
+  it("should return empty array when isInvisiblySmallElement is true", () => {
+    const rectElement = API.createElement({ type: "rectangle" });
+    mockSizeHelper.mockImplementation(() => true);
+
+    expect(restore.restoreElements([rectElement]).length).toBe(0);
+  });
+
+  it("should restore text element correctly passing value for each attribute", () => {
+    const textElement = API.createElement({
+      type: "text",
+      fontSize: 14,
+      fontFamily: 1,
+      text: "text",
+      textAlign: "center",
+      verticalAlign: "middle",
+      id: "id-text01",
+    });
+
+    const restoredText = restore.restoreElements([
+      textElement,
+    ])[0] as ExcalidrawTextElement;
+
+    expect(restoredText).toMatchSnapshot({
+      seed: expect.any(Number),
+    });
+  });
+
+  it("should restore text element correctly with unknown font family, null text and undefined alignment", () => {
+    const textElement = API.createElement({
+      type: "text",
+      textAlign: undefined,
+      verticalAlign: undefined,
+      id: "id-text01",
+    });
+
+    Object.defineProperty(textElement, "text", {
+      get: jest.fn(() => null),
+    });
+
+    Object.defineProperty(textElement, "font", {
+      get: jest.fn(() => "10 unknown"),
+    });
```

Comment:
you can pass these fields to `API.createElement` ?

Reply:
@ad1992 Unfortunately, the `API.CreateElement` does not accept the property `text` with `null` value. Also, it does not accept the property `font`. Therefore, I had to use jest to insert these parameters.